src/setup/sound.c

```diff
@@ -169,74 +168,43 @@ static void UpdateMidiDevice(TXT_UNCAST_ARG(widget), TXT_UNCAST_ARG(data))
 static txt_dropdown_list_t *MidiDeviceSelector(void)
 {
     txt_dropdown_list_t *result;
+    int num_devices;
     int all_devices;
-    int device_ids[MAX_MIDI_DEVICES];
-    MMRESULT mmr;
-    MIDIOUTCAPS mcaps;
     int i;
 
-    if (midi_num_devices > 0)
-    {
-        for (i = 0; i < midi_num_devices; ++i)
-        {
-            free(midi_names[i]);
-            midi_names[i] = NULL;
-        }
-        free(midi_names);
-        midi_names = NULL;
-    }
-    midi_num_devices = 0;
+    midi_index = 0;
+    num_devices = 1; // Always show MIDI_MAPPER.
+    all_devices = midiOutGetNumDevs(); // Does not include MIDI_MAPPER.
```

Comment:
`num_devices` and `all_devices` could be initialized as part of their definition for slightly shorter code.